lib/commits_loader.rb

```diff
@@ -0,0 +1,190 @@
+# frozen_string_literal: true
+
+require 'active_support'
+require 'active_support/core_ext/numeric'
+require 'active_support/core_ext/time'
+require 'json'
+require 'tempfile'
+require 'csv'
+require 'zlib'
+require 'pg'
+require_relative './github'
+require_relative './aws_helper'
+
+class CommitsLoader
+  HEADERS = %i[object_id repository message_headline message author_name authored_date committed_date pushed_date]
+  BUCKET = 'artsy-data'
+
+  def self.load_recent_commits
+    new.load_recent_commits
+  end
+
+  def initialize(since: 5.days.ago, _until: nil, org: "artsy")
+    @since = since
+    @until = _until
+    @org = org
+  end
+
+  def load_recent_commits
+    data = build_data
+    s3_object = upload_to_s3(data)
+    merge_into_warehouse(s3_object)
+  end
+
+  private
+
+  def build_data
+    data = []
+    each_recently_pushed_repo do |repo|
+      each_recently_pushed_commit(repo) do |commit|
+        data << {
+          object_id: commit.oid,
+          repository: repo.name,
+          message_headline: commit.messageHeadline,
+          message: commit.message,
+          author_name: commit.author.name,
+          authored_date: commit.authoredDate,
+          committed_date: commit.committedDate,
+          pushed_date: commit.pushedDate
+        }
+      end
+    end
+    data
+  end
+
+  def upload_to_s3(data)
+    key = "reports/engineering.commits/partial_#{Time.now.to_s(:number)}.csv.gz"
```

Comment:
Naming this "partial" to distinguish it from other "exports" that contain _all_ of the source data.

Reply:
Nope any single file on S3 will only have unique commits (including all from the designated time window, currently 5 days). But the same commit could, e.g., be in multiple days' files. As long as each file is processed via the merge operation (and uniquely identified by SHA), there shouldn't be duplicates anywhere.

Reply:
Sorry, poorly phrased questions. But this clarifies. I don't imagine this will be an issue, as I don't think there will be much use for the raw data other than to load into redshift. But just wanted to make sure I understood, correctly.